src/components/DocumentView/BlockContentRef.tsx

```diff
@@ -1,27 +1,77 @@
-import { DocumentBlockContentRef } from '@gitbook/api';
+import { CustomizationThemedURL, DocumentBlockContentRef } from '@gitbook/api';
 
 import { Card, Emoji } from '@/components/primitives';
+import { getSpaceCustomization, ignoreAPIError } from '@/lib/api';
+import { ResolvedContentRef } from '@/lib/references';
 
 import { BlockProps } from './Block';
+import { Image } from '../utils';
+
 
 export async function BlockContentRef(props: BlockProps<DocumentBlockContentRef>) {
     const { block, context, style } = props;
-    const kind = block?.data?.ref?.kind;
 
     const resolved = await context.resolveContentRef(block.data.ref, {
         resolveAnchorText: true,
     });
+
     if (!resolved) {
         return null;
     }
 
+    const kind = block?.data?.ref?.kind;
+    if (resolved.active && kind === 'space') {
+        return <SpaceRefCard {...props} resolved={resolved} />;
+    }
+
+    return (
+        <Card
+            leadingIcon={resolved.emoji ? <Emoji code={resolved.emoji} style="text-xl" /> : null}
+            href={resolved.href}
+            title={resolved.text}
+            style={style}
+        />
+    );
+}
+
+async function SpaceRefCard(props: { resolved: ResolvedContentRef } & BlockProps<DocumentBlockContentRef>) {
+    const { context, style, resolved } = props;
+    const spaceId = context.contentRefContext?.space.id;
+
+    if (!spaceId) { return null; }
+
+    const spaceCustomization = await ignoreAPIError(getSpaceCustomization(spaceId));
+    const customFavicon = spaceCustomization?.favicon;
+    const customEmoji = customFavicon && 'emoji' in customFavicon ? customFavicon.emoji : null;
+    const customIcon = customFavicon && 'icon' in customFavicon ? customFavicon.icon : null;
+
     return (
         <Card
-            leadingIcon={resolved.emoji ? <Emoji code={resolved.emoji} /> : null}
+            leadingIcon={customIcon ? <BlockContentRefIcon icon={customIcon} /> : customEmoji ? <Emoji code={customEmoji} style="text-xl" /> : null}
```

Comment:
If there's no icon or emoji, we fallback to null. That's different to `HeaderLogo` which falls back to the GitBook logo:

https://github.com/GitbookIO/gitbook/blob/main/src/components/Header/HeaderLogo.tsx#L114

Should we make sure that both act in the same way, so that the icon a user sees in the ContentRef card also matches the icon they'll see at the top of the page if they follow the link?

Reply:
Ok, I've made a `LogoIcon` that makes sure the logic is consistent between both. I included the customization emoji because it seems like it should show if you've selected that customization and not an icon image.